initrd/kraftignore.go

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+package initrd
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"kraftkit.sh/log"
+)
+
+// kraftignore filename
+const KraftignoreFileName = ".kraftignore"
+
+type IgnoringFileType string
+
+const (
+	Exist    = IgnoringFileType("Exist")
+	NotExist = IgnoringFileType("NotExist")
+	SkipDir  = IgnoringFileType("SkipDir")
+)
```

Comment:
Same here, we can declare these as errors:
```go
var (
  ErrFileExists    = fmt.Errorf("file exists")
  ErrFileNotExists = fmt.Errorf("filt does not exist")
  ErrIgnoreDir     = fmt.Errorf("ignored directory")
)
````

Reply:
You can then do comparisons via:
```go
if errors.Is(err, ErrFileExists)
```

Reply:
Surely, It can be done. But `kraftiignore.go` not uses these vars as errors, These vars are being used as notifiers for `filepath.WalkDir()` function in `directory.go`.

Please have an eye on `filepath.WalkDir()` function then lemme know if you really want me to change const vars as error vars :)